h2/src/main/org/h2/store/Data.java

```diff
@@ -655,6 +657,40 @@ public void writeValue(Value v) {
             }
             break;
         }
+        case Value.INTERVAL_YEAR:
+        case Value.INTERVAL_MONTH:
+        case Value.INTERVAL_DAY:
+        case Value.INTERVAL_HOUR:
+        case Value.INTERVAL_MINUTE: {
+            ValueInterval interval = (ValueInterval) v;
+            int ordinal = type - Value.INTERVAL_YEAR;
+            if (interval.isNegative()) {
+                ordinal = ~ordinal;
+            }
+            writeByte((byte) Value.INTERVAL_YEAR);
+            writeByte((byte) ordinal);
+            writeVarLong(interval.getLeading());
```

Comment:
if we encoding the "negative" boolean into the "leading" field (by just making leading negative), we can simplify all of this encoding/decoding and just write out the Value type directly, and reduce the overhead by one byte.

Reply:
Initially it was in leading field, but leading field can be zero. I tried to use special mark for it, but a lot of tricks were required, so I extracted it in a separate flag.

Reply:
For storage backends it's also better to have positive fields due to variable-length encoding that encodes negative values in inefficient way.

Reply:
You 're already clamping leading and negative so that negative zero  can never exist, so I don't see the problem?

Reply:
There is no problem because both fields are positive after ff41fdeb3c187101258b80b242ad1fbd340e0cec
Initial implementation had such bug.